### ISSUE TITLE
Dynniq EC2 sends old timestamps?

### ISSUE THREAD

**Issue by emiltin** [rsmp-nordic/rsmp_implementations#10]
At 2021-04-02 11:31:54,292 we receive:

{"mType":"rSMsg","type":"StatusUpdate","mId":"52df28e7-5163-4ae6-93a9-5e8405d4c429","ntsOId":"KK+AG1705=001TC000","xNId":"","cId":"KK+AG1705=001TC000","sTs":"**2021-04-02T09:31:54.262Z**","sS":[{"sCI":"S0001","n":"cyclecounter","s":"32","q":"recent"}]} 

Then at 2021-04-02 11:32:08,258 we receive:

{"mType":"rSMsg","type":"StatusUpdate","mId":"65379089-1da6-4533-bde4-8f2978482333","ntsOId":"KK+AG1705=001TC000","xNId":"","cId":"KK+AG1705=001TC000","sTs":"**2021-04-02T09:31:36.177Z**","sS":[{"sCI":"S0001","n":"cyclecounter","s":"46","q":"recent"}]} 
 
The second message has a timestamp that's about 18 seconds earlier that the first message.

Could it be the case that the second message was simply delayed in the network? I don't think so. Note that cycle counter is 32 in the first message, and 46 in the second, while the diff in timestamp is about 18 seconds. This seems to show that the either the timestamps or the cycle counter must be incorrect, since there’s no way the cycle counter can move from 46 to 32 in 18 seconds? The only way I can think of is if the clock, cycle time or offset was changed, which was not the case here.

**Comment by otterdahl** [comment]
We have seen similar issues with the Swarco controller as well. In that case it was due to several sources for time synchronization active at once. E.g. NTP + RSMP watchdog messages. This caused the controller to regularly to set time 1-2 seconds forwards and then backwards, causing the timestamps in status updates to sometimes jump backwards.

**Comment by emiltin** [comment]
I see. Does the issue still exists with Swarco TLCs?

**Comment by otterdahl** [comment]
It is still possible to define several sources for synchronization to be active at once. When you configure the TLC, one needs to be careful and not make mistakes. The NTP settings are separate from the RSMP settings, so it is not obvious for the user. As far as I know, there is no warning message either. The typical way to detect this problem is when the timestamps of StatusUpdates skips forwards and backwards.

**Comment by emiltin** [comment]
Not an optimal state of affairs.

**Comment by emiltin** [comment]
does the currect core draft include something about now allowing multiple time sources?
if NTP is active, i don't think it makes sense to set the clock via other sources?

**Comment by otterdahl** [comment]
> does the currect core draft include something about now allowing multiple time sources? if NTP is active, i don't think it makes sense to set the clock via other sources?

No, it doesn't mention multiple time sources, but perhaps more importantly; is the RSMP specification the correct place for such a requirement?

I think it is rather obvious that it doesn't make sense to use multiple time sources simultaneously. But I think it is more of a general requirement of the TLC, not specifically a RSMP requirement.

**Comment by emiltin** [comment]
many messages has some relation to the functionality of the equipment, eg. it must respond in a certain way if the equipment has support for x. you could say this is similar. if NTP is active, we could require that trying to set the clock should return e.g. a NotAck with "cannot change clock because it's controlled by NTP".
